gravityflow/gravityflow
# Fix mode selection on field filters with gravityflow_status_filter

## Description
Defining field filters with `any` mode is forced processed as `all`. In the case of multiple field filters defined, the `any` behaviour does not occur. The selections are only made in the `all` mode. This fix aims at addressing the issue, by either setting the mode to `any` or `all` (depending on the user selection). Alternatively, if nothing is specified, the mode selection defaults to `all`.

## Testing instructions
1. Create a snippet with gravityflow_status_filter containing a minimum of two field filters.
2. Ensure the mode is `any`.
3. Check the status page displays as expected with the assigned `any` mode.

## Automated Test Enhancements
N/A
 
## Screenshots
N/A

## Documentation Changes
Add an example of the filter page to show the usage of multiple field filters. 
```
add_filter( 'gravityflow_status_filter', 'sh_gravityflow_status_field_filter' );
function sh_gravityflow_status_field_filter( $filter_constraints ) {
	$filter_constraints = array(
		'status'		=> 'active',
		'field_filters'	=> array(
			'mode'		=> 'any',
			array(
				'key'		=> 'workflow_step',
				'operator'	=> 'in',
				'value'		=> array('34'),
			),
			array(
				'key'		=> 'workflow_final_status',
				'operator'	=> 'is',
				'value'		=> 'Approval',
			),
		),
	);
	return $filter_constraints;
}
```

The above snippet would display entries on the Status page such that it is either an Approved entry or it is an entry currently on Workflow Step 34.

## Checklist:
